### PR TITLE
Add Custom app ID

### PR DESCRIPTION
**Pull request by InfiniteCoder06**
Add Custom App ID generally used for Track Only Sources like ApkMirror
Without App ID
With Custom App ID

### CODE REVIEW

**Comment by ImranR98** [comment]
Not sure it's a good idea to let users change App IDs. They are meant to be unique and unchanging, and a lot of features have been written with this in mind (even app entry/config data is stored in a JSON file named after the app ID). Have you tested this thoroughly to make sure it doesn't break anything?

File: lib/pages/add_app.dart [hunk 1]
```diff
@@ -538,6 +538,22 @@ class AddAppPageState extends State<AddAppPage> {
...
+                  items: [
+                    [
+                      GeneratedFormTextField('appId',
+                          label: tr('appId'), required: false),
+                    ]
+                  ],
+                  onValueChanges: (values, valid, isBuilding) {
+                    if (!isBuilding) {
+                      setState(() {
+                        additionalSettings['appId'] = values['appId'];
```
Comment (ImranR98) [inline lib/pages/add_app.dart:553]: This value should be reset when the rest of the form is reset (when the picked source changes).

**Comment by ImranR98** [comment]
Still not sure about this. Will need to test it more.

**Comment by InfiniteCoder06** [comment]
Added Validation of App ID (Package Name) as well

> Still not sure about this. Will need to test it more.

**Comment by ImranR98** [comment]
It shouldn't be possible to use this when an app is not added as track-only (should be hidden). Also could you describe what ways you have tested this

**Pull request merged**